stubs/zxcvbn/zxcvbn/matching.pyi

```diff
@@ -0,0 +1,88 @@
+from collections.abc import Iterable, Mapping
+from decimal import Decimal
+from typing import Any, Pattern
+from typing_extensions import Literal, NotRequired, TypedDict
+
+from .adjacency_graphs import _Graph
+
+class _Match(TypedDict):
+    pattern: Literal["dictionary", "spatial", "repeat", "sequence", "regex", "date"]
+    token: str
+    i: int
+    j: int
+    guesses: int
+    guesses_log10: float
```

Comment:
`date_match()` is annotated as returning `list[_Match]` but the dictionaries in that list don't appear to have `guesses` or `guesses_log10` keys in them, so maybe these should be `NotRequired`?

Those dictionaries also have `year`, `month` and `day` keys, which don't currently appear in this `TypedDict`. https://github.com/dwolfhub/zxcvbn-python/blob/f596bef499c02bb6f0c8aef33b260364add2dc2b/zxcvbn/matching.py#L520-L529

And they sometimes have a `separator` key: https://github.com/dwolfhub/zxcvbn-python/blob/f596bef499c02bb6f0c8aef33b260364add2dc2b/zxcvbn/matching.py#L552

Reply:
I'll have a look later.